dali/pipeline/workspace/workspace.h

```diff
@@ -66,15 +68,17 @@ class ArgumentWorkspace {
     auto it = argument_inputs_.find(arg_name);
     DALI_ENFORCE(it != argument_inputs_.end(), "Argument \"" + arg_name + "\" not found.");
     if (it->second.should_update) {
-      // the underlying tensor list might have changed - update the views
-      it->second.tvec->UpdateViews();
+    // TODO(klecki): Remove when TensorList is replaced by proper the TensorBatch object
+      // the underlying tensor list might have changed - reshare the data
+      it->second.tvec->ShareData(*it->second.tlist);
     }
     return *it->second.tvec;
   }
 
  protected:
   struct ArgumentInputDesc {
-    shared_ptr<TensorVector<CPUBackend>> tvec;
+    std::shared_ptr<TensorVector<CPUBackend>> tvec;
+    std::shared_ptr<TensorList<CPUBackend>> tlist;
```

Comment:
Don't quite get why those are shared_ptr's

Reply:
nvm

Reply:
Those objects are shared between workspaces and realize the "data graph" for given iteration.